jdtls.ext/com.microsoft.jdtls.ext.core/src/com/microsoft/jdtls/ext/core/PackageCommand.java

```diff
@@ -367,17 +367,20 @@ private static IPackageFragmentRoot getPackageFragmentRootFromQuery(PackageParam
             return packageRoot;
         }
 
-        // jar in Referenced Libraries must be constructed from path
-        IJavaProject javaProject = getJavaProject(query.getProjectUri());
-        if (javaProject != null) {
-            try {
-                packageRoot = javaProject.findPackageFragmentRoot(Path.fromPortableString(query.getRootPath()));
-            } catch (JavaModelException e) {
-                return null;
+        if (query.getProjectUri() != null && query.getRootPath() != null) {
```

Comment:
just to check, is `query.getProjectUri()` cheap? if not, suggest to assign it to a var.

Reply:
Yes, they are just simple getters.